### DISCUSSION TITLE
DJI OSDK ROS does not work on Jetson AGX Xavier

### DISCUSSION THREAD

**Discussion by DevJadefly** [community/community#126602] [Programming Help]
### Body

Hello, 
I recently move to working on a Jetson AGX Xavier to develop on DJI M300 and I have some troubles. In fact, I have a windows computer linked to my drone to run DJI Simulator lite on DJI Assitant 2 and I have Onboard OSDK Ros  (v4.1) built on my jetson AGX Xavier, from where I launch dji_vehicle_node. I want to read data on the /dji_osdk_ros/local_position topic so I need GPS data
When I run telemetry_node it returns : 
[ INFO] [1717076452.731975330]: Currrent GPS health is 0
[ INFO] [1717076452.732103976]: Not enough GPS Satellites. 
[ INFO] [1717076452.732161355]: Cannot set Local Position reference
Except I have the simulator running so it is supposed to receive the fake gps from the simulation. 
I made it work once on the jetson and never again without changing anything. Does it sound familiar to you ? Have you any idea what would cause that ?
Thanks in advance !

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by nurulislam21** [comment]
This is a known intermittent issue with the OSDK simulator GPS bridge. A few things to check:

**1. Simulator link not fully established before node launch**
The most common cause of `GPS health: 0` with the simulator running is a timing issue — `dji_vehicle_node` initializes and polls GPS health before the simulator's fake GPS data has fully synced over the USB/serial link. Try adding a `rospy.sleep(5)` or a launch file delay before starting `telemetry_node`, or simply relaunch `dji_vehicle_node` after the simulator has been running for ~10 seconds.

**2. DJI Assistant 2 simulator must be actively running the simulation, not just open**
Make sure you hit **Start Simulation** inside DJI Assistant 2 — having it open on the simulator tab is not enough. The fake GPS is only broadcast once simulation is actively running.

**3. USB connection priority**
With the Jetson and a Windows PC both connected to the M300, the drone may be negotiating which host gets the data stream. Disconnect the Windows machine's data link (keep only the Jetson's OSDK USB connected) and retest. Two hosts competing for the OSDK serial channel can cause GPS to report 0 health sporadically.
